### PR TITLE
More bug fixes to legacy catalogs

### PR DESCRIPTION
**Pull request by yymao**
This PR contains a few more bug fixes to legacy catalogs:

 1. Remove `hsc-pdr1-xmm` since the files have been purged.
 2. Comment out unfinished modifiers in `dc2_dia_object.py` and fix a f-string bug.
 3. Check the availability of modelfit flux and flag in legacy object catalog.  
 4. Update path to instance catalog examples and fix a deprecation warning.

### CODE REVIEW

**Comment by yymao** [comment]
A full test run with this PR is done and all tests have passed.

File: GCRCatalogs/dc2_dia_object.py [hunk 1]
```diff
@@ -70,26 +70,26 @@ def _generate_modifiers(dm_schema_version=4, bands='ugrizy', **kwargs):  # pylin
...
+            # 'lcNonPeriodic': '',
+            # # These are possible to calculate if you require the Object Table
+            # 'nearbyObj': '',
+            # 'nearbyObjDist': '',
+            # 'nearbyObjLnP': '',
         }
 
-        modifiers['good'] = (create_basic_flag_mask,)
-        modifiers['clean'] = modifiers['good']
+        # modifiers['good'] = (create_basic_flag_mask, )
```
Comment (JoanneBogart) [inline GCRCatalogs/dc2_dia_object.py:91]: Why are these now commented out?  Should they have not been there in the first place?

File: GCRCatalogs/instance_catalog.py [hunk 2]
```diff
@@ -266,7 +266,7 @@ def _generate_native_quantity_list(self):
         return native_quantities
 
     def _pd_read_table(self, obj_type, **kwargs):
-        return pd.read_table(
+        return pd.read_csv(
```
Comment (JoanneBogart) [inline GCRCatalogs/instance_catalog.py:269]: From the documentation, it looks like read_csv and read_table behave identically when delim_whitespace is True.   Why did you change it?

File: GCRCatalogs/dc2_dia_object.py (hunk 1 above)
Comment (yymao) [reply GCRCatalogs/dc2_dia_object.py:91]: This line appears to be an unfinished line. There should be columns names following the `create_basic_flag_mask` function ([example](https://github.com/LSSTDESC/gcr-catalogs/blob/master/GCRCatalogs/dc2_object.py#L360)). 

However, I checked and could not find these columns needed to define `good` in the underlying file, so I commented out them for now. We'll need to check with @wmwv if we want to properly resolve this.

File: GCRCatalogs/instance_catalog.py (hunk 2 above)
Comment (yymao) [reply GCRCatalogs/instance_catalog.py:269]: `read_table` was deprecated at one point but apparently it has been "un-deprecated" now (see https://github.com/pandas-dev/pandas/issues/25220). 

The version of pandas that is currently in our `desc-stack` is not up-to-date enough so it still prints out a deprecation warning for `read_table`. However, it's true that this edit is not necessary.

Comment (JoanneBogart) [reply GCRCatalogs/instance_catalog.py:269]: Thanks; that explains it.

File: GCRCatalogs/dc2_dia_object.py (hunk 1 above)
Comment (JoanneBogart) [reply GCRCatalogs/dc2_dia_object.py:91]: Good enough for now.  Thanks.
